torch/csrc/jit/serialization/import.cpp

```diff
@@ -130,11 +156,14 @@ class ScriptModuleDeserializer final {
   IValue readArchive(const std::string& archive_name);
 
   std::shared_ptr<CompilationUnit> compilation_unit_;
-  std::shared_ptr<PyTorchStreamReader> reader_;
+  std::shared_ptr<PyTorchStreamReader> reader_concrete_;
```

Comment:
They have the same use purposes, but the old code differs in that it creates/owns the PyTorchStreamReader while the new code passes one in as a reference (the PackageImporter own the PyTorchStreamReader object). I wasn't sure how to have the ScriptModuleDeserializer handle both cases. I can change the "old" code to have the PyTorchStreamReader created in ExportModule or Module::Save upstream of ScriptModuleDeserializer and not have this confusion here

Reply:
Hm well, `shared_ptr` represents shared ownership of an object, so it could be used in the `PackageImporter` case as well, right? e.g. PackageImporter creates/owns a `shared_ptr<PyTorchStreamReader>`, and hands out a `shared_ptr` to `ScriptModuleDeserializer` for use.

Reply:
That works! I didn't know that you could get a shared_ptr from PyBind. I changed the implementation 👍